pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/ASTVariableDeclaratorId.java

```diff
@@ -71,28 +71,45 @@ public boolean isExplicitReceiverParameter() {
         return explicitReceiverParameter;
     }
 
+
+    /**
+     * Returns true if this declarator ID declares a formal
+     * parameter for a lambda with an inferred type, eg in
+     * {@code (a,b) -> a + b}. In that case, {@link #getTypeNode()}
+     * returns {@code null}, since the type node is absent.
+     */
+    public boolean isLambdaInferredFormalParameter() {
```

Comment:
I foresee the upcoming support for Java 10 local variables (which java 11 will extend to Lambdas) will need a VERY similar treatment. No `TypeNode` will be available, yet we should be able to infer a type for the variable itself.

It may be interesting to have this method named in a more future-safe fashion.

Reply:
Hum yes, that's a good catch. Maybe just "isTypeInferred" would be better then.